File: turingmachine.py
transitions = [[1,5,4,5,3], 
              [1,2,5,5,1],
              [2,5,5,0,2],
              [5,5,4,5,3],
              [4,4,4,4,4],
              [5,5,5,5,5]]

symbols = ["0","1"," ","x","y"]

pointer = [[1,1,0,1,1], [1,0,1,1,1],
           [0,1,1,1,0], 
           [1,1,0,1,1],
           [1,1,1,1,1],
           [1,1,1,1,1]]
           
replace = [[3,3,3,3,4], [0,4,3,3,4],
           [0,3,3,3,4],
           [0,1,2,3,4],
           [0,1,2,3,4],
           [0,1,2,3,4]]
           
state = 0

s = input("enter the string: ")
s = list(s + " ")

acc = 4
dec = 5 

tape = s
head = 0
i = 20

print("Tape: ", end="")
print(tape)
print("State = "+str(state) + ", Head = "+ str(head))

while state < 4:
  c = state
  
  symbol = symbols.index(tape[head]) # get index of symbol
  
  tape[head] = symbols[replace[c][symbol]] 
  state = transitions[c][symbol]
  
  if pointer[c][symbol] == 0:
    head -= 1
  else:
    head += 1
    
  print("Tape: ", end="")
  print(tape)
  
  print("State = "+str(state) + ", Head = "+ str(head))
  
  if state == 4:
    print("the given string passes")
  else:
    print("the given string does not pass")

File: turingmachine.java
import java.util.Arrays;
import java.util.Scanner;

public class TuringMachine {
    public static void main(String[] args) {
        int[][] transitions = {
            {1, 5, 4, 5, 3},
            {1, 2, 5, 5, 1},
            {2, 5, 5, 0, 2},
            {5, 5, 4, 5, 3},
            {4, 4, 4, 4, 4},
            {5, 5, 5, 5, 5}
        };

        String[] symbols = {"0", "1", " ", "x", "y"};

        int[][] pointer = {
            {1, 1, 0, 1, 1},
            {1, 0, 1, 1, 1},
            {0, 1, 1, 1, 0},
            {1, 1, 0, 1, 1},
            {1, 1, 1, 1, 1},
            {1, 1, 1, 1, 1}
        };

        int[][] replace = {
            {3, 3, 3, 3, 4},
            {0, 4, 3, 3, 4},
            {0, 3, 3, 3, 4},
            {0, 1, 2, 3, 4},
            {0, 1, 2, 3, 4},
            {0, 1, 2, 3, 4}
        };

        int state = 0;

        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String input = scanner.nextLine();
        input = input + " ";
        char[] tape = input.toCharArray();
        int head = 0;

        System.out.print("Tape: ");
        System.out.println(Arrays.toString(tape));
        System.out.println("State = " + state + ", Head = " + head);

        while (state < 4) {
            int c = state;
            int symbol = Arrays.asList(symbols).indexOf(String.valueOf(tape[head]));

            tape[head] = symbols[replace[c][symbol]].toCharArray()[0];
            state = transitions[c][symbol];

            if (pointer[c][symbol] == 0) {
                head -= 1;
            } else {
                head += 1;
            }

            System.out.print("Tape: ");
            System.out.println(Arrays.toString(tape));
            System.out.println("State = " + state + ", Head = " + head);

            if (state == 4) {
                System.out.println("The given string passes");
            } else {
                System.out.println("The given string does not pass");
            }
        }
    }
}
